rcl/include/rcl/node_type_cache.h

```diff
@@ -0,0 +1,63 @@
+// Copyright 2023 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__NODE_TYPE_CACHE_H_
+#define RCL__NODE_TYPE_CACHE_H_
+
+#include "rcl/node.h"
+#include "rcl/types.h"
+#include "rcl/visibility_control.h"
+#include "rosidl_runtime_c/message_type_support_struct.h"
+#include "rosidl_runtime_c/type_description/type_description__struct.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct rcl_node_type_cache_type_info_t {
```

Comment:
Maybe we  just call this  `rcl_type_info_t` so that it sounds more generic for use outside the caching context.